### PR TITLE
Jetpack AI: Only display "turn list into table" menu item for top level lists

### LINKED ISSUE #40137: Do not display "Turn List Into Table" item when we're not focused on the top level list

### PR DESCRIPTION
**Pull request by mwatson**
Fixes #40137

## Proposed changes:

* Currently the "Turn List Into Table" menu item is displayed for sub-lists, however the transform action does not work for sub-lists (also it would cause us to insert a table into a list, which is illegal)
* This removes that option in the drop down unless we are in the parent list

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A but this was mentioned in the [previous PR](https://github.com/Automattic/jetpack/pull/40148).

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

Once you've checked out the branch and built Jetpack, you will need to enable the feature via a snippet or similar means:

```
define( 'JETPACK_BLOCKS_VARIATION', 'beta' );
add_filter( 'list_to_table_transform_enabled', '__return_true' );
```

Once that's done, create a new post and create a list with multiple sub-lists, like so:

Screenshot 2024-11-13 at 5 22 24 PM

Validate:

* Upon clicking the AI Assistant icon on the root list you should see an option in the dropdown that says "Turn list into a table"
* If you click the AI Assistant icon at any other level of the list the option should not be present.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/components/ai-assistant-toolbar-dropdown/dropdown-content.tsx [hunk 1]
```diff
@@ -191,12 +194,16 @@ type AiAssistantToolbarDropdownContentProps = {
...
 	blockType,
+	clientId,
 	disabled = false,
 	onAskAiAssistant,
 	onRequestSuggestion,
 }: AiAssistantToolbarDropdownContentProps ): ReactElement {
 	const blockQuickActions = quickActionsList[ blockType ] ?? [];
 
+	const { getBlockParents } = select( 'core/block-editor' );
+	const blockParents = getBlockParents( clientId );
```
Comment (mwatson) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/components/ai-assistant-toolbar-dropdown/dropdown-content.tsx:208]: The other option we can do is to always display the transform option and just find the root list and transform the entire thing every time, regardless of where they click AI Assistant.

Comment (CGastrell) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/components/ai-assistant-toolbar-dropdown/dropdown-content.tsx:208]: This sounds interesting, do you think is doable without refactoring too much? Can we explore it? UX wise I think it'd be much more comfortable for the user

**Review by CGastrell — DISMISSED** [review]
I like the idea of automating the trigger from root of list, but I'll leave it up for your consideration. Otherwise, this worked fine for me!

**Comment by mwatson** [comment]
Adding missing property to `AiAssistantToolbarDropdownContentProps` which I think will fix the JS tests issue. Going to see if I can make it so it always transforms the top level list as discussed. Will timebox it for a few hours just in case.

**Review by CGastrell — APPROVED** [review]
LGTM!

**Pull request merged**